tools/FileTestDefs.sh

```diff
@@ -38,9 +38,34 @@ staged_grep() {
 # Utility function for reporters that enables lots of decorators in grep
 # Works like staged_grep
 pretty_grep() {
-    echo -n -e "\033[0;35m${@: -1}\033[0m:"
-    git show ":./${@: -1}" | \
-        GREP_COLOR='1;37;41' grep -n $color_option "${@:1:$(($#-1))}"
+    local -a args
+    local file file_prefix
+    while [ $# -gt 0 ] ; do
```

Comment:
this would be easier to understand if you added a comment explaining that this loop accumulates all of the dash-preceded flags, followed by the grep pattern, and that following the loop the only arguments remaining in $@ will be file names.